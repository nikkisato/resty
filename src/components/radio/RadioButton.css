.RadioButton{
    label{
        padding: 0.4rem;
    }
    &:hover {
        background-color:rgba(0,0,255,0.5)
    }
}



input:checked + label {
    background-color: red;
    color: white;

    transition: ease;
}