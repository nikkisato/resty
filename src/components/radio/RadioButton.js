import React from 'react';
import PropTypes from 'prop-types';
import styles from './RadioButton.css';

const RadioButton = ({ method, label, value, onChange }) => (
  <div className={styles.RadioButton}>
    <input 
      id={styles.RadioButton} 
      type="radio" 
      name='method' 
      value={value} 
      onChange={onChange}/>
    <label htmlFor={value}>{label}</label>
  </div>
);

RadioButton.propTypes = {
  method: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};


export default RadioButton;
