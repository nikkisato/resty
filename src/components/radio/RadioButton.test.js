import React from 'react';
import { shallow } from 'enzyme';
import RadioButton from './RadioButton';

describe('RadioButton component', () => {
  it('renders RadioButton', () => {
    const wrapper = shallow(<RadioButton />);
    expect(wrapper).toMatchSnapshot();
  });
});
