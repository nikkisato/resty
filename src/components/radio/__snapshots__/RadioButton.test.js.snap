// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton component renders RadioButton 1`] = `
<div
  className="RadioButton"
>
  <input
    id="RadioButton"
    name="method"
    type="radio"
  />
  <label />
</div>
`;
